contribute/design/openebs-maya-exporter-deep-visibility.md

```diff
@@ -0,0 +1,227 @@
+
+6. ISCSI Connection: (istgt_iscsi_execute)
+    - Login request count : No of logins attempted to target (iscsi_op_login)
+    - Logout request count: No of logout attempted to target (iscsi_op_logout)
+    - Discovery request count: No of discovery requests (iscsiadm) ()
+    - Login Request duration (Histogram): Duration took to process the login request.
+    - Lu workers count: No of lu worker thread  (spec->luworkers)
+    - Used capacity: Used capacity of volume
+7. IO counters:  (replicate)
+    - Read10 count
+    - Read12 count
+    - Read16 count
+    - …. All the scsi opcodes
+8. Q count: 
+    - Ready
+    - Blocked
+    - … etc
+
+ ## Schema
+ ```
+ {
+  "loginReqCnt": "6",
+  "loginReqSuccessCnt": "6",
+  "logoutReqCnt": "5",
+  "logoutReqSuccessCnt": "5",
+  "discoveryReqCnt": "6",
+  "discoveryReqSuccessCnt": "6",
+  "read6Cnt": "0",
+  "read10Cnt": "749",
+  "read12Cnt": "0",
+  "read16Cnt": "0",
+  "write6Cnt": "0",
+  "write10Cnt": "1122",
+  "write12Cnt": "0",
+  "write16Cnt": "0",
+  "sync10Cnt": "70",
+  "sync16Cnt": "0",
+  "luWorkers": 6,
+  "snapshotCreateReqCnt": "321",
+  "snapshotPrepareReqCnt": "1",
+  "snapshotCreateReqSuccessCnt": "47",
+  "snapshotPrepareReqSuccessCnt": "1",
+  "startRebuildReqCnt": "4",
+  "startRebuildReqSuccessCnt": "0",
+  "prepareRebuildReqCnt": "4",
+  "prepareRebuildReqSuccessCnt": "0",
+  "maintCmdQLen": "0",
+  "maintBlockedQLen": "0",
+  "cmdQLen": "26",
+  "blockedQLen": "0",
+  "replicas": [
+    {
+      "replicaId": "6162",
+      "Address": "127.0.0.1",
+      "Mode": "Healthy",
+      "checkpointedIOSeq": "1000",
+      "inflightRead": "0",
+      "inflightWrite": "0",
+      "inflightSync": "0",
+      "quorum": "1",
+      "upTime": 379,
+      "waitQ": 0,
+      "readyQ": 0,
+      "blockedQ": 0,
+      "snapshotCreateReqCnt": "81",
+      "snapshotPrepareReqCnt": "1",
+      "snapshotCreateReqSuccessCnt": "64",
+      "snapshotPrepareReqSuccessCnt": "1",
+      "startRebuildReqCnt": "1",
+      "startRebuildReqSuccessCnt": "1",
+      "prepareRebuildReqCnt": "2",
+      "prepareRebuildReqSuccessCnt": "1"
+    },
+    {
+      "replicaId": "6163",
+      "Address": "127.0.0.1",
+      "Mode": "Healthy",
+      "checkpointedIOSeq": "1000",
+      "inflightRead": "0",
+      "inflightWrite": "0",
+      "inflightSync": "0",
+      "quorum": "1",
+      "upTime": 377,
+      "waitQ": 0,
+      "readyQ": 0,
+      "blockedQ": 0,
+      "snapshotCreateReqCnt": "81",
+      "snapshotPrepareReqCnt": "0",
+      "snapshotCreateReqSuccessCnt": "61",
+      "snapshotPrepareReqSuccessCnt": "0",
+      "startRebuildReqCnt": "1",
+      "startRebuildReqSuccessCnt": "1",
+      "prepareRebuildReqCnt": "1",
+      "prepareRebuildReqSuccessCnt": "1"
+    },
+    {
+      "replicaId": "6161",
+      "Address": "127.0.0.1",
+      "Mode": "Degraded",
+      "checkpointedIOSeq": "1000",
+      "inflightRead": "0",
+      "inflightWrite": "6",
```

Comment:
Also add `volsize`.

Reply:
replica version

Reply:
@mynktl REPLICA_VERSION is a header defined in zrepl_prot.h, so i what is the benefit of using this ?

Reply:
Yes @utkarshmani1997 , It is defined in zrepl_prot.h. It will help us to check if there is any mismatch in replica and target version.